### DISCUSSION TITLE
How to use columns with json arrays?

### DISCUSSION THREAD

**Discussion by Igorgro** [grafana/grafana-infinity-datasource#449] [Q&A] [answered]
Let's say, I want to visualize the following JSON(just a minimal working example) using backend parser:
```json
{
  "values": {
    "a": [1, 2, 3, 4, 5],
    "b": [54, 613, 121, 44, 67]
  }
}
```
I have the following columns setup:
![image](https://user-images.githubusercontent.com/22894169/200630929-7237fb5d-c723-47e8-9733-448e5226143a.png)

But this is what I'm getting (table view):
![image](https://user-images.githubusercontent.com/22894169/200631112-682bdea8-5d92-4fd6-9a19-07ca367bc335.png)

Looks like the Infinity datasource returns both arrays as strings which makes it impossible to visualize them. What I'm trying to get is the following (this is simulated by creating two requests and applying "concatenate" transformation to them):

![image](https://user-images.githubusercontent.com/22894169/200632047-9678ce0f-9433-4e63-9e17-61ba3b672577.png)

Am I misunderstand something?

**Comment by yesoreyeram** [comment, marked as answer]
if the built-in parser doesn't support your need, use UQL+JSONata to get the data want. [example](https://community.grafana.com/t/how-can-i-calculate-and-add-a-column-of-percentage/75296/6).

**Reply by Igorgro** [reply]
I'm sorry but what is built-in parser suitable for? I can remeber that I was able to visualize JSONs like that without using UQL in the version 1.0.0.

**Reply by yesoreyeram** [reply]
default parser will nicely fit for various flat data structures. If you need complex data manipulation according to your API response, use UQL/JSONata to achieve what you want. version doesn't matter.

**Reply by yesoreyeram** [reply]
backend parser is in early stages, more parsing options coming soon.

**Reply by Igorgro** [reply]
Thank you for you answers. I switched to default parser and enabled "Is data in columnar format?", this made parser works as expected.
